nuxt-community/axios-module
# fix: preserve default headers with custom headers

There is a breaking change with axios [axios/axios@v0.19.2...v0.20.0](https://github.com/axios/axios/compare/v0.19.2...v0.20.0) that when passing headers config when calling `axios`, it discards `defaults.headers.common` (used for both config and `setHeader` utility.

resolves #411 

**Update:** Axios regression found by @1ed regarding defaults overriding: https://github.com/axios/axios/pull/2844 <3

---

Test flow for #411:

- `/api/cookie` endpoint that simply returns value of `mycookie` header if set
- Make a random string in client-side and set cookie "`mycookie`"
- Make SSR request
  - SSR request makes a call to `/api/cookie` in `asyncData` using `$axios` to check what headers are passed to API when using axios in server side
- Value is checked against initial random one (and passing)

### Locally trying:

 - Clone repo and run `yarn install`
 - Switch to `fix/411` branch
 - Run `yarn dev`
 - Go to http://localhost:3000/cookie
